### PR TITLE
feat: add CloudFront distribution for API

### PR DESCRIPTION
**Pull request by patheard**
# Summary
This will be used to cache GET/HEAD requests (respecting response cache control headers) and to redirect HTTP requests to HTTPS.

# Expected changes
Add CloudFront distribution and S3 logging bucket.

### CODE REVIEW

File: terragrunt/env/dev/cloudfront/terragrunt.hcl [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+inputs = {
+  certificate_arn = dependency.domain.outputs.certificate_arn
+  domain_name     = dependency.domain.outputs.api_gateway_domain_name
+  origin_id       = dependency.domain.outputs.api_gateway_id
+}
+
+terraform {
+  source = "../../../aws//cloudfront"
+}
```
Comment (ikennaokpala) [inline terragrunt/env/dev/cloudfront/terragrunt.hcl:28]: New line?

File: terragrunt/aws/cloudfront/inputs.tf [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+variable "domain_name" {
+  description = "(Required) domain name to use for CloudFront"
+  type        = string
+}
+
+variable "origin_id" {
+  description = "(Required) ID of the CloudFront Origin"
+  type        = string
+}
```
Comment (ikennaokpala) [inline terragrunt/aws/cloudfront/inputs.tf:14]: New Line?

File: terragrunt/env/dev/cloudfront/terragrunt.hcl (hunk 1 above)
Comment (patheard) [reply terragrunt/env/dev/cloudfront/terragrunt.hcl:28]: Sure, at the end of the file right?

Comment (ikennaokpala) [reply terragrunt/env/dev/cloudfront/terragrunt.hcl:28]: Yes Pat

**Review by timarney — APPROVED** [review]

**Pull request merged**